### DISCUSSION TITLE
Writing in word posting to quarto blog - where to put the images?

### DISCUSSION THREAD

**Discussion by davidleitch1** [quarto-dev/quarto-cli#8286] [Q&A] [answered]
### Description

Hi I write in word and program in Python. My articles go another site but I wish to publish my own copy on my personal site using quarto. I could use pandoc but the beginner approach is to save word docx as .html and copy the output into body of a index.qmd file. Word saves the images into a directory in the same directory as the .html output. I would just like to know where I should put that director so that the index.qmd can find the images. I apologise for such a beginner question and look forward to a long association with quarto.

**Comment by rgaiacs** [comment, marked as answer]
Dear @davidleitch1,

You can convert your Microsoft Office Word (`.docx`) file to Markdown using Pandoc. Consider 
[mwe.docx](https://github.com/quarto-dev/quarto-cli/files/13947026/mwe.docx) that has a 1x1 pixel image, `pandoc --from docx --to markdown --standalone mwe.docx` will produce

```
---
author: Jane Doe
title: My Title
---

Lorem ipsum

![](media/image1.png){width="3.3333333333333335e-3in"
height="3.3333333333333335e-3in"}
```

that you can save into a file and copy the `media/image1.png` to your project.

The advantage of convert to Markdown is that you get the YAML header.

You can also convert your Microsoft Office Word (`.docx`) file to HTML as you are doing but you will not have the YAML header. If you convert to HTML, you can use the argument `--embed` and have the image inside the HTML. For example, `pandoc --from docx --to html --embed mwe.docx` will produce

```
<p>Lorem ipsum</p>
<p><img src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAIAAACQd1PeAAABhWlDQ1BJQ0MgcHJvZmlsZQAAKJF9kT1Iw1AUhU9TpUUqCnYQEcxQnSyIijhKFYtgobQVWnUweekfNGlIUlwcBdeCgz+LVQcXZ10dXAVB8AfE2cFJ0UVKvC8ptIjxwuN9nHfP4b37AKFRYarZNQGommWk4jExm1sVA6/woR9BCBiRmKkn0osZeNbXPXVT3UV5lnffn9Wr5E0G+ETiOaYbFvEG8cympXPeJw6zkqQQnxOPG3RB4keuyy6/cS46LPDMsJFJzROHicViB8sdzEqGSjxNHFFUjfKFrMsK5y3OaqXGWvfkLwzltZU012kNI44lJJCECBk1lFGBhSjtGikmUnQe8/APOf4kuWRylcHIsYAqVEiOH/wPfs/WLExNukmhGND9Ytsfo0BgF2jWbfv72LabJ4D/GbjS2v5qA5j9JL3e1iJHQN82cHHd1uQ94HIHGHzSJUNyJD8toVAA3s/om3LAwC3Qs+bOrXWO0wcgQ7NavgEODoGxImWve7w72Dm3f3ta8/sBJSpyiBCpwRoAAAAJcEhZcwAALiMAAC4jAXilP3YAAAAHdElNRQfoARAIAwPnjR6hAAAAGXRFWHRDb21tZW50AENyZWF0ZWQgd2l0aCBHSU1QV4EOFwAAAAxJREFUCNdjYDD/DwABcQE3puMJygAAAABJRU5ErkJggg==" /></p>
```

I hope this help you.

**Reply by davidleitch1** [reply]
Dear @rgaiacs 

Thankyou. You have set me on a good path I think. The embed method is quite slow with the images employed in my test article and for some reason the first two images (mac screen shots copied from Excel) did not render although others did. So I plan to do more work with the markdown version. I guess will need to copy all the images out of the word version and save them as the appropriate image names in the appropriate quarto sub directory.

----
Later: the markdown option worked well. Next problem is some #403 errors when configuring the blog with holoviz panel.

**Reply by rgaiacs** [reply]
Dear @davidleitch1,

> The embed method is quite slow

To convert from Word to HTML? Or for your web browser to render the HTML?

> for some reason the first two images (mac screen shots copied from Excel) did not render

Hard to say way without having access to your `.docx` file. Some image formats are not supported by web browsers, for example `TIFF`. **Pandoc will keep the images intact**. For example, given [mwe-with-tiff.docx](https://github.com/quarto-dev/quarto-cli/files/13960970/mwe-with-tiff.docx) that has a 1x1 pixel TIFF image, the output of `pandoc --from docx --to html --wrap=none mwe-with-tiff.docx ` is

```
<p>MWE with TIFF</p>
<p>Lorem ipsum.</p>
<p><img src="media/image1.tiff" style="width:1.04167in;height:1.04167in" /></p>
```

The browser will not be able to display the TIFF file.

Another thing to take into consideration is that web browsers have different limits for `data:image/png;base64`. https://stackoverflow.com/a/41755526/1802726 states that Chrome has a 2MB limit. **Try reduce the size of your images.**
